### PR TITLE
fix(react-tree): Apply aria-selected prop to aria-selected instead of aria-expanded

### PR DESCRIPTION
**Pull request by sopranopillow**
PR to fix wrong placement of aria-selected prop. 

## Related Issue(s)

- Fixes #31020

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">33</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">37</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/Avatar-mount_base.html">615</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/Avatar-mount.html">647</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/Button-mount_base.html">285</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/Button-mount.html">297</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/Field-mount_base.html">1139</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/Field-mount.html">1149</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/FluentProvider-mount_base.html">723</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/FluentProvider-mount.html">686</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/FluentProviderWithTheme-mount_base.html">80</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/FluentProviderWithTheme-mount.html">78</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">33</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">37</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">75</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">78</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/MakeStyles-mount_base.html">845</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/MakeStyles-mount.html">863</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/Persona-mount_base.html">1749</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/Persona-mount.html">1699</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/SpinButton-mount_base.html">1405</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/SpinButton-mount.html">1358</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/SwatchPicker-mount_base.html">1480</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31117/perf-test-react-components/SwatchPicker-mount.html">1521</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-tree/src/components/TreeItem/useTreeItem.tsx [hunk 1]
```diff
@@ -241,8 +241,8 @@ export function useTreeItem_unstable(props: TreeItemProps, ref: React.Ref<HTMLDi
         'aria-level': level,
         'aria-checked': selectionMode === 'multiselect' ? checked : undefined,
         // Casting: when selectionMode is 'single', checked is a boolean
-        'aria-selected': selectionMode === 'single' ? !!checked : undefined,
-        'aria-expanded': ariaSelected !== undefined ? ariaSelected : itemType === 'branch' ? open : undefined,
+        'aria-selected': ariaSelected !== undefined ? ariaSelected : selectionMode === 'single' ? !!checked : undefined,
+        'aria-expanded': itemType === 'branch' ? open : undefined,
```
Comment (YuanboXue-Amber) [inline packages/react-components/react-tree/src/components/TreeItem/useTreeItem.tsx:245]: `aria-expanded` from user prop will be overridden.
```suggestion
        'aria-expanded': props['aria-expanded'] ?? (itemType === 'branch' ? open : undefined),
```

Comment (sopranopillow) [reply packages/react-components/react-tree/src/components/TreeItem/useTreeItem.tsx]: added in https://github.com/microsoft/fluentui/pull/31117/commits/8b4337c108d7367763b3356fb15b3bbed0ddd571

**Review by YuanboXue-Amber — APPROVED** [review]

**Pull request merged**